src/components/alarm/AlarmController.cpp

```diff
@@ -79,15 +74,25 @@ void AlarmController::ScheduleAlarm() {
   tmAlarmTime->tm_isdst = -1; // use system timezone setting to determine DST
 
   // now can convert back to a time_point
-  alarmTime = std::chrono::system_clock::from_time_t(std::mktime(tmAlarmTime));
-  auto mSecToAlarm = std::chrono::duration_cast<std::chrono::milliseconds>(alarmTime - now).count();
+  return std::chrono::system_clock::from_time_t(std::mktime(tmAlarmTime));
+}
+
+void AlarmController::ScheduleAlarm() {
+  // Determine the next time the alarm needs to go off and set the app_timer
+  app_timer_stop(alarmAppTimer);
+
+  auto mSecToAlarm = std::chrono::duration_cast<std::chrono::milliseconds>(CalculateAlarmTimePoint() - dateTimeController.CurrentDateTime()).count();
   app_timer_start(alarmAppTimer, APP_TIMER_TICKS(mSecToAlarm), this);
 
   state = AlarmState::Set;
 }
 
 uint32_t AlarmController::SecondsToAlarm() {
-  return std::chrono::duration_cast<std::chrono::seconds>(alarmTime - dateTimeController.CurrentDateTime()).count();
+  if (state == AlarmState::Set) {
+    return std::chrono::duration_cast<std::chrono::seconds>(CalculateAlarmTimePoint() - dateTimeController.CurrentDateTime()).count();
+  } else {
+    return std::chrono::duration_cast<std::chrono::seconds>(CalculateAlarmTimePoint() - dateTimeController.CurrentDateTime()).count();
+  }
```

Comment:
This if statement isn't doing anything anymore.

Reply:
Fixed